File: app/src/main/java/com/datarockets/mnchkn/utils/LogUtil.java
package com.datarockets.mnchkn.utils;

public class LogUtil {
    public static final String LOG_PREFIX = "mnchkn ";
    public static final int LOG_PREFIX_LENGTH = LOG_PREFIX.length();
    public static final int MAX_LOG_TAG_LENGTH = 23;
    
    private LogUtil() {}

    public static String makeLogTag(String str) {
        if (str.length() > MAX_LOG_TAG_LENGTH - LOG_PREFIX_LENGTH) {
            return LOG_PREFIX + str.substring(0, MAX_LOG_TAG_LENGTH - LOG_PREFIX_LENGTH - 1);
        }
        return LOG_PREFIX + str;
    }

    public static String makeLogTag(Class cls) {
        return makeLogTag(cls.getSimpleName());
    }

}

File: app/src/main/kotlin/com/datarockets/mnchkn/utils/LogUtil.kt
package com.datarockets.mnchkn.utils

import kotlin.reflect.KClass

object LogUtil {

    val LOG_PREFIX = "mnchkn "
    val LOG_PREFIX_LENGTH = LOG_PREFIX.length
    val MAX_LOG_TAG_LENGTH = 23

    fun makeLogTag(str: String): String {
        if (str.length > MAX_LOG_TAG_LENGTH - LOG_PREFIX_LENGTH) {
            return LOG_PREFIX + str.substring(0, MAX_LOG_TAG_LENGTH - LOG_PREFIX_LENGTH - 1)
        }
        return LOG_PREFIX + str
    }

    fun makeLogTag(cls: KClass<*>): String {
        return makeLogTag(cls.simpleName!!)
    }

}
